FHEM/14_SD_WS.pm

```diff
@@ -1,4 +1,4 @@
-# $Id: 14_SD_WS.pm 21666 2022-01-30 10:19:30Z elektron-bbs $
+# $Id: 14_SD_WS.pm v3.5.4 2022-01-30 10:19:30Z elektron-bbs $
```

Comment:
Ich weiß nicht, ob das gut ist, in den Modulen die Versionsnummer zu ändern.
Die Ausgabe von "version" in FHEM kommt damit offensichtlich nicht so richtig klar:

```
00_FBAHAHTTP.pm        25767 2022-03-03 18:52:56Z rudolfkoenig
10_FBDECT.pm           25069 2021-10-13 05:58:51Z rudolfkoenig
14_SD_WS.pm            21666 2022-01-30 10:19:30Z elektron-bbs
# $Id: 00_SIGNALduino.pm v3.5.4 2022-03-12 17:55:30Z sidey79 $
# $Id: SD_ProtocolData.pm 3.5.x 2022-01-30 10:19:30Z elektron-bbs $
```

Reply:
Was willst Du stattdessen dort hinterlegen?

Die SVN commit ID ist ja falsch

Reply:
Entweder gar nichts daran ändern, oder vielleicht statt "v3.5.4" nur "354".

Reply:
Das soll ja den Versionsstand der Datei darstellen.
Es bei der SVN commit ID zu belassen ist dadurch falsch.

354 ist ebenfalls eine gültige SVN Referenzen.

Vielleicht könnten wir einen Patch für die Anzeige vorschlagen?

Reply:
So wichtig erscheint mir das nicht. Mir ist eben bisher nur aufgefallen, das wir da aus der Reihe tanzen.
@HomeAutoUser und ich hatten irgendwann mal angefangen bei 0. Ich habe dann dort ab und an mal hochgezählt:

```
14_SD_AS.pm              350 2020-10-01 20:16:11Z elektron-bbs
14_SD_BELL.pm              0 2021-07-12 22:30:35Z HomeAuto_User
10_SD_GT.pm                1 2020-05-25 21:00:00Z elektron-bbs
88_SIGNALduino_TOOL.pm     0 2021-10-12 07:58:00Z HomeAuto_User
```

Anhand der relativ kleinen Zahlen erkennen wir, das die Dateien nicht aus SVN stammen.

Reply:
Wir können das Format nicht einhalten, solange wir dir Datei nicht im SVN einchecken, da der Wert nur dann korrekt gesetzt wird, wenn ein commit durchgeführt wird.

Den Mehrwert, valide aber nicht zutreffende SVN commit IDs zu hinterlegen verstehe ich nicht. Dadurch wird der Eintrag doch falsch

Reply:
https://github.com/fhem/fhem-mirror/blob/6bccfb00851718c0c70cf5df5b0a0ad024f0f9be/fhem/FHEM/98_version.pm#L67

Wir müssten hier nur eine Änderung der wirken, dass auch Werte <> Zahlen an der betreffenden Stelle erlaubt sind

Reply:
Ich habe die Regex angepasst

https://regex101.com/r/HUdUmu/1

Ich werde mal testen wie es damit aussieht

Reply:
Ergebnis sieht ganz gut aus für den Anfang,

Patch ist eingereicht:
https://forum.fhem.de/index.php?topic=126908.new#new

Reply:
Das zieht sich jetzt schon seit über 3 Wochen hin. Wollen wir nicht erstmal den Pull request übernehmen?
Ich hätte mittlerweile weitere Änderungen für das Modul in petto.
Die Anpassung der Versionsnummer kannst du ja auch später noch vornehmen.

Reply:
@elektron-bbs 

Wir können den PR so mergen aus meiner Sicht, die nicht optimale Anzeige des version Befehls verschwindet, sobald der Patch angenommen wurde.